src/back/glsl/mod.rs

```diff
@@ -2333,14 +2356,22 @@ impl<'a, W: Write> Writer<'a, W> {
                         write!(self.out, ")",)?;
                     }
                     crate::ImageQuery::NumLayers => {
-                        let fun_name = match class {
-                            ImageClass::Sampled { .. } | ImageClass::Depth { .. } => "textureSize",
-                            ImageClass::Storage { .. } => "imageSize",
+                        let (fun_name, lod) = match class {
```

Comment:
we should better use the existing helpers `class.is_multisampled()` for consistency